src/ORM/Table.php

```diff
@@ -1479,6 +1484,8 @@ public function save(EntityInterface $entity, $options = [])
         ]);
 
         if ($entity->errors()) {
+            $this->dispatchEvent('Model.failedSave', compact('entity', 'options'));
```

Comment:
I'm not sure we should do a check this early. This one means probably that validation failed. And this failure should be handled in client code or using `Marshaller` event-cycle.